### PR TITLE
Integrate pre-commit and update files

### PR DESCRIPTION
**Pull request by vulder**

### CODE REVIEW

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5482658&bid=17966584)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5482658&bid=17966584)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           

Clones added
============
- varats/projects/c_projects/gzip.py  1
- varats/projects/c_projects/lrzip.py  4
- varats/projects/c_projects/glibc.py  4
- varats/projects/c_projects/libvpx.py  4
- varats/projects/c_projects/openssl.py  4
- varats/projects/c_projects/xz.py  3
- varats/projects/c_projects/tmux.py  4
- varats/projects/c_projects/coreutils.py  3
- varats/projects/c_projects/gravity.py  3
- varats/projects/c_projects/busybox.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5482658&bid=17966584)

File: tests/paper/test_artefacts.py [hunk 1]
```diff
@@ -5,8 +5,7 @@
 from pathlib import Path
 from tempfile import NamedTemporaryFile
 
-from varats.paper.artefacts import load_artefacts_from_file, ArtefactType, \
-    PlotArtefact
+from varats.paper.artefacts import ArtefactType, PlotArtefact, load_artefacts_from_file
```
Comment (boehmseb) [inline tests/paper/test_artefacts.py]: This line is too long.

File: tests/test_vara_manager.py [hunk 2]
```diff
@@ -5,8 +5,7 @@
 import unittest
 from pathlib import Path
 
-from varats.vara_manager import LLVMProjects, VaRAProjectsIter,\
-    VaRAExtraProjectsIter
+from varats.vara_manager import LLVMProjects, VaRAExtraProjectsIter, VaRAProjectsIter
```
Comment (boehmseb) [inline tests/test_vara_manager.py]: This line is too long.

File: varats/experiments/vara_fc_taint_analysis.py [hunk 3]
```diff
@@ -7,19 +7,18 @@
...
+from benchbuild.settings import CFG as BB_CFG
+from benchbuild.utils.cmd import FileCheck, echo, rm
+from plumbum import ProcessExecutionError
+
 from varats.data.report import FileStatusExtension as FSE
-from varats.utils.experiment_util import (exec_func_with_pe_error_handler,
-                                          PEErrorHandler)
+from varats.data.reports.taint_report import TaintPropagationReport as TPR
+from varats.experiments.vara_full_mtfa import VaRATaintPropagation
+from varats.utils.experiment_util import PEErrorHandler, exec_func_with_pe_error_handler
```
Comment (boehmseb) [inline varats/experiments/vara_fc_taint_analysis.py]: This line is too long.

File: varats/gui/views/cr_bar_view.py [hunk 4]
```diff
@@ -4,14 +4,13 @@
...
 
-from PyQt5.QtWidgets import QWidget, QFileDialog, QMessageBox
 from PyQt5.QtCore import Qt, QTimer
+from PyQt5.QtWidgets import QFileDialog, QMessageBox, QWidget
 
-from varats.gui.views.ui_CRBarView import Ui_Form
 from varats.data.data_manager import VDM
-from varats.data.reports.commit_report import (CommitReport, CommitReportMeta,
-                                               CommitMap)
+from varats.data.reports.commit_report import CommitMap, CommitReport, CommitReportMeta
```
Comment (boehmseb) [inline varats/gui/views/cr_bar_view.py]: This line is too long.

File: varats/paper/case_study.py [hunk 5]
```diff
@@ -2,30 +2,32 @@
...
-from benchbuild.project import Project
-from scipy.stats import halfnorm
 import numpy as np
 import pygit2
+from benchbuild.project import Project
+from scipy.stats import halfnorm
 
-from varats.data.provider.release.release_provider import (ReleaseType,
-                                                           ReleaseProvider)
+from varats.data.provider.release.release_provider import ReleaseProvider, ReleaseType
```
Comment (boehmseb) [inline varats/paper/case_study.py]: Line is too long.

File: varats/paper/paper_config_manager.py [hunk 6]
```diff
@@ -4,22 +4,20 @@
...
 
-from varats.data.revisions import get_all_revisions_files
-from varats.tools.commit_map import create_lazy_commit_map_loader
-
+import varats.paper.paper_config as PC
 from varats.data.report import FileStatusExtension, MetaReport
-from varats.paper.case_study import (CaseStudy,
-                                     get_newest_result_files_for_case_study)
+from varats.data.revisions import get_all_revisions_files
+from varats.paper.case_study import CaseStudy, get_newest_result_files_for_case_study
```
Comment (boehmseb) [inline varats/paper/paper_config_manager.py]: Line is too long.

File: varats/plots/blame_lorenz_curve.py [hunk 7]
```diff
@@ -10,14 +10,12 @@
 import numpy as np
 import pandas as pd
 
-from varats.data.databases.blame_interaction_database import (
-    BlameInteractionDatabase)
+from varats.data.databases.blame_interaction_database import BlameInteractionDatabase
```
Comment (boehmseb) [inline varats/plots/blame_lorenz_curve.py]: Line is too long.

File: varats/plots/blame_lorenz_curve.py [hunk 8]
```diff
@@ -10,14 +10,12 @@
...
 from varats.data.reports.commit_report import CommitMap
 from varats.paper.case_study import CaseStudy
 from varats.plots.plot import Plot
-from varats.plots.repository_churn import (draw_code_churn,
-                                           build_repo_churn_table)
-from varats.utils.git_util import calc_repo_code_churn, ChurnConfig
+from varats.plots.repository_churn import build_repo_churn_table, draw_code_churn
```
Comment (boehmseb) [inline varats/plots/blame_lorenz_curve.py]: Line is too long.

File: varats/plots/commit_interactions.py [hunk 9]
```diff
@@ -11,9 +11,8 @@
 import pandas as pd
 from matplotlib import cm
 
-from varats.data.databases.commit_interaction_database import (
-    CommitInteractionDatabase)
-from varats.paper.case_study import (CaseStudy, CSStage)
+from varats.data.databases.commit_interaction_database import CommitInteractionDatabase
```
Comment (boehmseb) [inline varats/plots/commit_interactions.py]: Line is too long.

File: varats/projects/c_projects/x264.py [hunk 10]
```diff
@@ -6,24 +6,20 @@
...
 from benchbuild.utils.download import with_git
 from benchbuild.utils.run import run
-
 from plumbum import local
 
-from varats.utils.project_util import get_all_revisions_between
-
 from varats.paper.paper_config import project_filter_generator
-from varats.utils.project_util import wrap_paths_to_binaries
+from varats.utils.project_util import get_all_revisions_between, wrap_paths_to_binaries
```
Comment (boehmseb) [inline varats/projects/c_projects/x264.py]: Line is too long.

File: varats/tools/research_tools/vara.py [hunk 11]
```diff
@@ -2,25 +2,27 @@
...
+from plumbum.cmd import ln, mkdir
+from PyQt5.QtCore import QProcess
 
+from varats.plots.plot_utils import check_required_args
 from varats.settings import CFG, save_config
-from varats.tools.research_tools.research_tool import (ResearchTool, CodeBase,
-                                                       SubProject)
-from varats.vara_manager import (BuildType, run_process_with_output,
-                                 set_vara_cmake_variables, ProcessManager)
+from varats.tools.research_tools.research_tool import CodeBase, ResearchTool, SubProject
```
Comment (boehmseb) [inline varats/tools/research_tools/vara.py]: Line is too long.

File: varats/tools/driver_build_setup.py [hunk 12]
```diff
@@ -2,22 +2,24 @@
...
+import typing as tp
 from pathlib import Path
 
 from PyQt5.QtCore import Qt
 from PyQt5.QtWidgets import QApplication
 
 from varats.gui.buildsetup_window import BuildSetup
-from varats.settings import get_value_or_default, CFG, save_config
+from varats.settings import CFG, get_value_or_default, save_config
+from varats.tools.research_tools.research_tool import ResearchTool, SpecificCodeBase
```
Comment (boehmseb) [inline varats/tools/driver_build_setup.py]: Line is too long.

File: varats/vara_manager.py [hunk 13]
```diff
@@ -5,23 +5,20 @@
...
 from threading import RLock
 
-from PyQt5.QtCore import (QRunnable, QThreadPool, pyqtSlot, pyqtSignal, QObject,
-                          QProcess)
-
-from plumbum import local, TF, RETCODE
-from plumbum.cmd import git, mkdir, ln, grep, cmake
+from plumbum import RETCODE, TF, local
+from plumbum.cmd import cmake, git, grep, ln, mkdir
+from PyQt5.QtCore import QObject, QProcess, QRunnable, QThreadPool, pyqtSignal, pyqtSlot
```
Comment (boehmseb) [inline varats/vara_manager.py]: Line is too long.

**Review by boehmseb — CHANGES_REQUESTED** [review]
There seems to be an issue with the import formatter that causes lines to exceed the 80-char limit. I have marked these lines with comments.

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
           

Clones added
============
- varats/projects/c_projects/gravity.py  3
- varats/projects/c_projects/lrzip.py  4
- varats/projects/c_projects/libvpx.py  4
- varats/projects/c_projects/glibc.py  4
- varats/projects/c_projects/openssl.py  4
- varats/projects/c_projects/busybox.py  5
- varats/projects/c_projects/tmux.py  4
- varats/projects/c_projects/coreutils.py  3
- varats/projects/c_projects/xz.py  3
- varats/projects/c_projects/gzip.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5482658&bid=17966584)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
           

Clones added
============
- varats/projects/c_projects/openssl.py  4
- varats/projects/c_projects/coreutils.py  3
- varats/projects/c_projects/glibc.py  4
- varats/projects/c_projects/gravity.py  3
- varats/projects/c_projects/busybox.py  5
- varats/projects/c_projects/lrzip.py  4
- varats/projects/c_projects/xz.py  3
- varats/projects/c_projects/gzip.py  1
- varats/projects/c_projects/tmux.py  4
- varats/projects/c_projects/libvpx.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5482658&bid=17966584)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
           

Clones added
============
- varats/projects/c_projects/lrzip.py  4
- varats/projects/c_projects/tmux.py  4
- varats/projects/c_projects/gzip.py  1
- varats/projects/c_projects/glibc.py  4
- varats/projects/c_projects/xz.py  3
- varats/projects/c_projects/libvpx.py  4
- varats/projects/c_projects/openssl.py  4
- varats/projects/c_projects/busybox.py  5
- varats/projects/c_projects/coreutils.py  3
- varats/projects/c_projects/gravity.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5482658&bid=17966584)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
           

Clones added
============
- varats/projects/c_projects/tmux.py  4
- varats/projects/c_projects/coreutils.py  3
- varats/projects/c_projects/gravity.py  3
- varats/projects/c_projects/busybox.py  5
- varats/projects/c_projects/gzip.py  1
- varats/projects/c_projects/lrzip.py  4
- varats/projects/c_projects/libvpx.py  4
- varats/projects/c_projects/openssl.py  4
- varats/projects/c_projects/xz.py  3
- varats/projects/c_projects/glibc.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5482658&bid=17966584)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 8
           

Clones added
============
- varats/projects/c_projects/lrzip.py  4
- varats/projects/c_projects/libvpx.py  4
- varats/projects/c_projects/gzip.py  1
- varats/experiments/commit_report_experiment.py  1
- varats/projects/c_projects/openssl.py  4
- varats/experiments/just_compile.py  1
- varats/projects/c_projects/tmux.py  4
- varats/projects/c_projects/xz.py  3
- varats/projects/c_projects/glibc.py  4
- varats/projects/c_projects/gravity.py  3
- varats/projects/c_projects/busybox.py  5
- varats/projects/c_projects/coreutils.py  3
- varats/experiments/phasar_env_analysis.py  2
- varats/experiments/vara_full_mtfa.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5482658&bid=17966584)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 8
           

Clones added
============
- varats/projects/c_projects/libvpx.py  4
- varats/projects/c_projects/tmux.py  4
- varats/projects/c_projects/gravity.py  3
- varats/projects/c_projects/coreutils.py  3
- varats/projects/c_projects/gzip.py  1
- varats/experiments/just_compile.py  1
- varats/experiments/phasar_env_analysis.py  2
- varats/experiments/commit_report_experiment.py  1
- varats/projects/c_projects/xz.py  3
- varats/experiments/vara_full_mtfa.py  2
- varats/projects/c_projects/busybox.py  5
- varats/projects/c_projects/glibc.py  4
- varats/projects/c_projects/openssl.py  4
- varats/projects/c_projects/lrzip.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5482658&bid=17966584)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
           

Clones added
============
- varats/projects/c_projects/tmux.py  4
- varats/projects/c_projects/gravity.py  3
- varats/projects/c_projects/openssl.py  4
- varats/projects/c_projects/gzip.py  1
- varats/projects/c_projects/xz.py  3
- varats/projects/c_projects/glibc.py  4
- varats/projects/c_projects/busybox.py  5
- varats/experiments/commit_report_experiment.py  1
- varats/experiments/vara_full_mtfa.py  2
- varats/projects/c_projects/libvpx.py  4
- varats/projects/c_projects/lrzip.py  4
- varats/experiments/phasar_env_analysis.py  2
- varats/projects/c_projects/coreutils.py  3
- varats/experiments/just_compile.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5482658&bid=17966584)

**Review by boehmseb — APPROVED** [review]

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
           

Clones added
============
- varats/projects/c_projects/gzip.py  1
- varats/projects/c_projects/busybox.py  5
- varats/projects/c_projects/xz.py  3
- varats/projects/c_projects/openssl.py  4
- varats/projects/c_projects/lrzip.py  4
- varats/projects/c_projects/glibc.py  4
- varats/experiments/commit_report_experiment.py  1
- varats/projects/c_projects/coreutils.py  3
- varats/projects/c_projects/libvpx.py  4
- varats/experiments/vara_full_mtfa.py  2
- varats/experiments/phasar_env_analysis.py  2
- varats/experiments/just_compile.py  1
- varats/projects/c_projects/gravity.py  3
- varats/projects/c_projects/tmux.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5482658&bid=17966584)

**Pull request merged**